prove/Develop05/ProgressionGoal.cs

```diff
@@ -44,20 +52,28 @@ public override bool CheckCompletionStatus()
 
     public override void DisplayProgressMessage()
     {
-        if (_progress == _duration)
+        if (CheckCompletionStatus() && _progress > _duration)
         {
-            Console.WriteLine($"Level {_currentLevel} completed.");
-            _progress = 0;
-            _totPoint = _bonusPoints;
-            Spin();
+            _totPoint = 0;
         }
-
-        if (CheckCompletionStatus())
+        else
         {
-            _totPoint = 0; // Reset value to zero
-            Console.WriteLine($"You have completed all {_totalLevel} levels of the {_name} goal. You rock!");
-            _totPoint = _bonusPoints + _completionBonus;
-            Spin();
+            if (_progress == _duration)
+            {
+                Console.WriteLine($"Level {_currentLevel} completed.");
+                _progress = 0;
+                _totPoint = _bonusPoints;
+                Spin();
+            }
+
+            if (CheckCompletionStatus())
+            {
+                _totPoint = 0; // Reset value to zero
+                Console.WriteLine($"You have completed all {_totalLevel} levels of the {_name} goal. You rock!");
+                _totPoint = _bonusPoints + _completionBonus;
+                _progress = (_duration += 1);
+                Spin();
```

Comment:
I don't love the spinner here.  It makes things take longer for testing.

Reply:
Spinner duration adjusted